src/ServiceControl.UnitTests/Monitoring/HeartbeatTypesArrayToInstanceConverterTests.cs

```diff
@@ -0,0 +1,172 @@
+                        "Endpoint": "SampleEndpoint",
+                        "StartedAt": "2022-12-01T12:00:00Z",
+                        "HostProperties": {
+                            "Property1": "Value1",
+                            "Property2": "Value2"
+                        },
+                        "HostDisplayName": "SampleHostDisplayName",
+                        "Host": "SampleHost"
+                    }
+                ]
+                """, options);
+
+            Assert.IsNotNull(endpointStartup);
+            Assert.AreEqual(new Guid("3fa85f64-5717-4562-b3fc-2c963f66afa6"), endpointStartup.HostId);
+            Assert.AreEqual("SampleEndpoint", endpointStartup.Endpoint);
+            Assert.AreEqual(new DateTime(2022, 12, 1, 12, 0, 0, 0, System.DateTimeKind.Utc), endpointStartup.StartedAt);
+            Assert.AreEqual("SampleHostDisplayName", endpointStartup.HostDisplayName);
+            Assert.AreEqual("SampleHost", endpointStartup.Host);
+            CollectionAssert.AreEqual(new Dictionary<string, string>
+            {
+                { "Property1", "Value1" },
+                { "Property2", "Value2" }
+            }, endpointStartup.HostProperties);
+        }
+
+        [Test]
+        public void Should_deserialize_single_register_endpoint_startup()
+        {
+            // sample json for RegisterEndpointStartup
+            var endpointStartup = JsonSerializer.Deserialize<RegisterEndpointStartup>("""
+                {
+                    "HostId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
+                    "Endpoint": "SampleEndpoint",
+                    "StartedAt": "2022-12-01T12:00:00Z",
+                    "HostProperties": {
+                        "Property1": "Value1",
+                        "Property2": "Value2"
+                    },
+                    "HostDisplayName": "SampleHostDisplayName",
+                    "Host": "SampleHost"
+                }
+                """, options);
+
+            Assert.IsNotNull(endpointStartup);
+            Assert.AreEqual(new Guid("3fa85f64-5717-4562-b3fc-2c963f66afa6"), endpointStartup.HostId);
+            Assert.AreEqual("SampleEndpoint", endpointStartup.Endpoint);
+            Assert.AreEqual(new DateTime(2022, 12, 1, 12, 0, 0, 0, System.DateTimeKind.Utc), endpointStartup.StartedAt);
+            Assert.AreEqual("SampleHostDisplayName", endpointStartup.HostDisplayName);
+            Assert.AreEqual("SampleHost", endpointStartup.Host);
+            CollectionAssert.AreEqual(new Dictionary<string, string>
+            {
+                { "Property1", "Value1" },
+                { "Property2", "Value2" }
+            }, endpointStartup.HostProperties);
+        }
+
+        [Test]
+        public void Should_deserialize_register_potentially_missing_heartbeat_arrays()
+        {
+            var potentiallyMissingHeartbeats = JsonSerializer.Deserialize<RegisterPotentiallyMissingHeartbeats>("""
```

Comment:
Like the string literals. Thanks for doing that. I should have done that myself. Thanks for taking it one step further